### PR TITLE
Global layout component

### PR DESCRIPTION
**Pull request by aquibbaig**
App.js needs to be singular and free of all nested components, only subscribed to global stores and persist configurations. Common components for each page should come under a global layout component. 

Signed-off-by: Aquib Baig <aquibbaig97@gmail.com>

### CODE REVIEW

File: dashboard/v1.0/src/components/layouts/GlobalLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import Header from './Header';
+import Sidebar from './Sidebar';
+
+export default props => {
```
Comment (ganeshpatro321) [inline dashboard/v1.0/src/components/layouts/GlobalLayout.tsx]: props keyword here can kind of mislead newbies to react. We can change it to the component name maybe. @aquibbaig

File: dashboard/v1.0/.prettierrc.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
         {
             "files": [".prettierrc", ".babelrc", ".eslintrc", ".stylelintrc"],
             "options": {
-                "parser": "json"
+                "parser": "json",
+                "tabWidth": 2
```
Comment (Harkishen-Singh) [inline dashboard/v1.0/.prettierrc.json]: Please make sure that this has effect in the CI tests. They should run strictly against these rules and fail on a single out. Else, this would be pointless.

File: dashboard/v1.0/src/App.tsx [hunk 3]
```diff
@@ -1,20 +1,13 @@
...
-        <Sidebar />
-      </div>
-      <div className="inner-component">
-        <Header />
-        <Navigator />
-      </div>
-    </div>
+    <GlobalLayout>
+      <Navigator />
+    </GlobalLayout>
```
Comment (Harkishen-Singh) [inline dashboard/v1.0/src/App.tsx]: Could you please tell why is this necessary?
Also, `GlobalLayout` should be changed to something more relatable like `Wrap` or something related to the topic. _(I am not good at naming)_

File: dashboard/v1.0/src/components/layouts/GlobalLayout.tsx (hunk 1 above)
Comment (Harkishen-Singh) [reply dashboard/v1.0/src/components/layouts/GlobalLayout.tsx]: Agreed. How about `root`? as a root component in the react-dom.

Comment (aquibbaig) [reply dashboard/v1.0/src/components/layouts/GlobalLayout.tsx]: Thanks @ganeshpatro321!

**Comment by aquibbaig** [comment]
@Harkishen-Singh,
> App.js needs to be singular and free of all nested components, only subscribed to global stores and persist configurations. Common components for each page should come under a global layout component. 

Having a layout simplifies much of our codebase, freeing App.js only to things that it needs. We may need to add other components such as ContainerQuery, Persistors, etc. It's a common practice for `React` developers.

File: dashboard/v1.0/src/App.tsx (hunk 3 above)
Comment (aquibbaig) [reply dashboard/v1.0/src/App.tsx]: It's a layout that uses composition. So `GlobalLayout` is perfect. We can use `BasicLayout` otherwise.

Comment (Harkishen-Singh) [reply dashboard/v1.0/src/App.tsx]: I just want to avoid `Global` from naming. Why not give the name related to something you want to implement?

Comment (aquibbaig) [reply dashboard/v1.0/src/App.tsx]: `BasicLayout` or `DefaultLayout` will be good?

Comment (aquibbaig) [reply dashboard/v1.0/src/App.tsx]: `BasicLayout` or `DefaultLayout` will be good?

Comment (aquibbaig) [reply dashboard/v1.0/src/App.tsx]: `BasicLayout` or `DefaultLayout` will be good?

Comment (aquibbaig) [reply dashboard/v1.0/src/App.tsx]: cool

File: dashboard/v1.0/src/components/layouts/BaseLayout.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import Header from './Header';
+import Sidebar from './Sidebar';
+
+export default function BaseLayout(props) {
+  return (
+    <>
```
Comment (Harkishen-Singh) [inline dashboard/v1.0/src/components/layouts/BaseLayout.tsx]: We can remove these. Its being handled by `div`

File: dashboard/v1.0/src/components/layouts/BaseLayout.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      <div className="App">
+        <div className="sidebar">
+          <Sidebar />
+        </div>
+        <div className="inner-component">
+          <Header />
+          {props.children}
+        </div>
+      </div>
+    </>
```
Comment (Harkishen-Singh) [inline dashboard/v1.0/src/components/layouts/BaseLayout.tsx]: same.
